src/main/webapp/script.js

```diff
@@ -18,21 +18,196 @@ var arrayOfProducts = Array();
 async function fetchDataAsync() {
     const response = await fetch('test.html');
     arrayOfProducts = await response.json();
```

Comment:
Why are we loading all the products data on page load? Shouldn't we only get the product types and tags here and only during filtering fetch the products.

Reply:
I'm not entirely sure what this comment means.

Reply:
Let's say we have 100 products which belong to 2 categories - lipsticks and mascara. There are two ways to filter this data -
Option 1 - Make a server call to fetch all 100 products during the page load. Then when a user clicks on lipsticks filter, we show only lipsticks by applying filtering logic client side (in javascript).
Option 2 - Make a server call to fetch the 2 categories (lipsticks and mascara) on page load. Then when a user clicks on lipsticks filter, we make a server call to fetch all products with lipstick as product type. 

We are doing Option 1 now, I am suggesting Option 2.